google/cloud/bigquery/table.py

```diff
@@ -2142,7 +2149,7 @@ def __hash__(self):
         return hash(self._key())
 
     def __repr__(self):
-        key_vals = ["{}={}".format(key, val) for key, val in self._key()]
+        key_vals = ["{}={}".format(key, repr(val)) for key, val in self._key()]
```

Comment:
The side effect of the `repr()` is that numeric values are now presented as strings, which breaks one of the unit tests:
```py
>>> TimePartitioning(expiration_ms=42)                                                                                                                                                                                                                                                                                                                                                                
TimePartitioning(expiration_ms='42',type_='DAY')
```
Even though the property still works as expected, because it's converted from/to string on get/set, we should really fix that here in the `__repr__()`, as those internal conversions should be considered an implementation detail.

Reply:
Yes I eventually call `repr` for non built-in type objects only in the `_key` method.